routes/nft.go

```diff
@@ -656,36 +645,46 @@ func (fes *APIServer) GetNFTShowcase(ww http.ResponseWriter, req *http.Request)
 			dropNumToFetch := dropEntry.DropNumber - 1
 			dropEntry, err = fes.GetNFTDropEntry(dropNumToFetch)
 			if err != nil {
-				_AddInternalServerError(ww, fmt.Sprintf(
-					"GetNFTShowcase: Problem getting drop #%d: %v", dropNumToFetch, err))
-				return
+				return nil, fmt.Errorf(
+					"GetNFTShowcase: Problem getting drop #%d: %v", dropNumToFetch, err)
 			}
 		}
 	}
+	return dropEntry, nil
+}
 
-	// Now that we have the drop entry, fetch the NFTs.
+func (fes *APIServer) GetNFTDropEntryByIdx(dropIdx int) (_dropEntry *NFTDropEntry, _err error) {
+	dropEntry, err := fes.GetNFTDropEntry(uint64(dropIdx))
+	if err != nil {
+		return nil, fmt.Errorf(
+			"GetNFTShowcase: Problem getting drop #%d: %v", dropIdx, err)
+	}
+	return dropEntry, nil
+}
+
+func (fes *APIServer) FetchNFTSForDrop(ReaderPublicKeyBase58Check string, readerPublicKeyBytes []byte, dropEntry *NFTDropEntry) (_nftCollectionResponses []*NFTCollectionResponse, _err error) {
 	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
 	if err != nil {
-		_AddBadRequestError(ww, fmt.Sprintf("GetNFTShowcase: Error getting utxoView: %v", err))
-		return
+		return nil, fmt.Errorf(
+			"GetNFTShowcase: Error getting utxoView: %v", err)
 	}
 
 	verifiedMap, err := fes.GetVerifiedUsernameToPKIDMap()
 	if err != nil {
-		_AddBadRequestError(ww, fmt.Sprintf("GetNFTBidsForNFTPost: Error getting verified user map: %v", err))
-		return
+		return nil, fmt.Errorf(
+			"GetNFTBidsForNFTPost: Error getting verified user map: %v", err)
 	}
 
 	var readerPKID *lib.PKID
-	if requestData.ReaderPublicKeyBase58Check != "" {
+	if ReaderPublicKeyBase58Check != "" {
```

Comment:
I don't think we need to pass the ReaderPublicKeyBase58Check in here. we can instead check that the length of the readerPublicKeyBytes is correct.

`if len(readerPublicKeyBytes) == btcec.PubKeyBytesLenCompressed`

let's remove ReaderPublicKeyBase58Check from this funciton